### PR TITLE
projects:evb_discovery_firmware: Added support for STM32H5

### PR DESCRIPTION
**Pull request by jansunil**
1) Added support for STM32H5 platform in the existing code

2) Added app_config_stm32.h/app_config_stm32.c files

3) Updated gitignore and added mbedignore

4) Added a macro for unused device ID in the app_config_mbed.h

### CODE REVIEW

**Comment by jansunil** [comment]
Changelog: Updated src.mk to exclude stm32 specific files for mbed build

**Comment by jansunil** [comment]
Hi All- are there any review comments on this PR?

File: projects/evb_discovery_firmware/app/app_config.c
Comment (ribdp) [inline projects/evb_discovery_firmware/app/app_config.c:1]: Should we update the copyright year on this file, app_config.h and app_config_mbed.h?

**Review by ribdp — APPROVED** [review]

File: projects/evb_discovery_firmware/STM32/.extSettings [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+app/libraries/precision-converters-library/board_info/=../../../libraries/precision-converters-library/board_info/board_info.c;../../../libraries/precision-converters-library/board_info/board_info.h;
+
+app/libraries/no-OS/drivers/eeprom/24xx32a/=../../../libraries/no-OS/drivers/eeprom/24xx32a/24xx32a.c;../../../libraries/no-OS/drivers/eeprom/24xx32a/24xx32a.h;
+
+app/libraries/no-OS/=../../../libraries/no-OS/util/;../../../libraries/no-OS/include;
+
+app/libraries/no-OS/drivers/platform/stm32/=../../../libraries/no-OS/drivers/platform/stm32/stm32_uart.c;../../../libraries/no-OS/drivers/platform/stm32/stm32_uart.h;../../../libraries/no-OS/drivers/platform/stm32/stm32_i2c.h;../../../libraries/no-OS/drivers/platform/stm32/stm32_i2c.c;
+
+app/libraries/no-OS/iio/=../../../libraries/no-OS/iio/iio.c;../../../libraries/no-OS/iio/iio.h;../../../libraries/no-OS/iio/iiod.h;../../../libraries/no-OS/iio/iiod.c;../../../libraries/no-OS/iio/iio_trigger.c;;../../../libraries/no-OS/iio/iio_trigger.h;
```
Comment (SaikiranGudla) [inline projects/evb_discovery_firmware/STM32/.extSettings:17]: nit: Are iio_trigger files used here?

File: projects/evb_discovery_firmware/app/app_config.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 /* List of supported platforms */
 #define	MBED_PLATFORM		1
+#define STM32_PLATFORM      2
```
Comment (SaikiranGudla) [inline projects/evb_discovery_firmware/app/app_config.h:28]: nit: Can add space for uniform alignment.

File: projects/evb_discovery_firmware/STM32/.extSettings (hunk 1 above)
Comment (jansunil) [reply projects/evb_discovery_firmware/STM32/.extSettings:17]: No, this has been removed now

**Comment by jansunil** [comment]
@SaikiranGudla @ribdp -  Do these changes look good for a merge?

**Review by SaikiranGudla — APPROVED** [review]

**Review by ribdp — APPROVED** [review]

**Pull request merged**